src/compas/geometry/brep/brep.py

```diff
@@ -0,0 +1,410 @@
+from compas.geometry import Geometry
+from compas.plugins import pluggable
+
+
+@pluggable(category="factories")
+def new_brep(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_mesh(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_box(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_cylinder(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_sphere(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_cone(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_surface(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+class BRep(Geometry):
+    """Class for Boundary Representation of geometric entities.
+
+    Attributes
+    ----------
+    vertices : list[:class:`~compas_occ.brep.BRepVertex`], read-only
+        The vertices of the BRep.
+    edges : list[:class:`~compas_occ.brep.BRepEdge`], read-only
+        The edges of the BRep.
+    loops : list[:class:`~compas_occ.brep.BRepLoop`], read-only
+        The loops of the BRep.
+    faces : list[:class:`~compas_occ.brep.BRepFace`], read-only
+        The faces of the BRep.
+    frame : :class:`~compas.geometry.Frame`, read-only
+        The local coordinate system of the BRep.
+    area : float, read-only
+        The surface area of the BRep.
+    volume : float, read-only
+        The volume of the regions contained by the BRep.
+
+    Other Attributes
+    ----------------
+    occ_shape : ``TopoDS_Shape``
```

Comment:
this one probably doesn't belong here

Reply:
tricky one, this would make pythonocc a core dependency, making a massive CAD kernel a core dependency...
its a nice contribution ( have devoted a **lot** of time and energy on pythonocc ) but better make the dependency explicit and have this kind of functionality baked in `compas_occ`?

Reply:
👍  I guess the contained shape attribute is not part of the interface and instead is something specific to each implementation

Reply:
well, an underlying native representation will be present in both cases. in case of nurbs, this is currently referred to as, for example, `occ_curve` and `rhino_curve`, which turns out to be a bit annoying. perhaps better would be to have both refer to an underlying `_curve`.

same could be said about the breps...

Reply:
@brgcode so from the import statement it would be obvious whether the `brep` backend is `occ` || `rhino`?
cool & ambitious to get `brep` support 🦾💎

Reply:
no this would work in the same way as with the nurbs plugins. the constructor methods will hand off the actual construction process to a pluggable. the pluggable will receive plugins from `compas_rhino` and `compas_occ`. depending on the execution context, the plugin manager will then decide which plugin to use...

Reply:
As @gonzalocasas points out, I think it's worth discussing if this is part of the public interface, named one way or another. 

The main reason I kept it decorated with `property` is to allow replacing (or lazy setting) the native type after instantiation. The setter then already extracts the `data`. That way the other accessors are not dependent on the native instance being present, which is sometimes difficult due to reason mentioned in https://github.com/compas-dev/compas/pull/1057#discussion_r920874977.

Having said that, I guess there's no technical reason not to prefix the property's name with `_`, don't know how conventional that is though. Alternatively, could have an explicit `_set_native_instance()`.

Should the underlying native instance be in any way public?